### PR TITLE
84240 shared key authentication

### PR DESCRIPTION
**Pull request by Vishal14583**
1. Added shared key authentication for ERP facade endpoints.
2. Functional Tests updated accordingly.
3. Unit test cases covered.

### CODE REVIEW

**Review by RaviK2233 — APPROVED** [review]
Reviewed FT and looks good

File: tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/PriceChangeScenario.cs [hunk 1]
```diff
@@ -5,10 +5,10 @@
 namespace UKHO.ERPFacade.API.FunctionalTests.FunctionalTests
 {
     [TestFixture]
-    public class BulkPriceUpdateScenarios
+    public class PriceChangeScenarios
     {
 
-        private BulkPriceUpdateEndpoint _bulkPriceUpdate { get; set; }
+        private PriceChangeEndpoint _PriceChange { get; set; }
```
Comment (harshal11869) [inline tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/PriceChangeScenario.cs:11]:
```suggestion
        private PriceChangeEndpoint _priceChange { get; set; }
```

File: tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs [hunk 2]
```diff
@@ -10,7 +10,6 @@ namespace UKHO.ERPFacade.API.FunctionalTests.Helpers
     public class WebhookEndpoint
     {
         private readonly RestClient client;
```
Comment (harshal11869) [inline tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/WebhookEndpoint.cs:12]:
```suggestion
        private readonly RestClient _client;
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/PriceChangeScenario.cs (hunk 1 above)
Comment (MitaliSingh16) [reply tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/PriceChangeScenario.cs]: Done

**Review by shrey1501620 — APPROVED** [review]

**Review by martyn-fewtrell — COMMENTED** [review]
Looks good

**Review by martyn-fewtrell — APPROVED** [review]

**Review by jrippington — APPROVED** [review]
Looks good

**Pull request merged**